python/benchmark/benchmark/bench_dbscan.py

```diff
@@ -0,0 +1,283 @@
+#
+# Copyright (c) 2022-2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pprint
+import time
+from typing import Any, Dict, Iterator, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+from pyspark.ml.feature import VectorAssembler
+from pyspark.ml.functions import array_to_vector, vector_to_array
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import array, col, sum
+from pyspark.sql.types import DoubleType, StructField, StructType
+
+from .base import BenchmarkBase
+from .utils import inspect_default_params_from_func, with_benchmark
+
+
+class BenchmarkDBSCAN(BenchmarkBase):
+    def _supported_class_params(self) -> Dict[str, Any]:
+        from pyspark.ml.clustering import KMeans
+
+        params = inspect_default_params_from_func(
+            KMeans.__init__,
+            [
+                "distanceMeasure",
+                "featuresCol",
+                "labelCol",
+                "predictionCol",
+                "probabilityCol",
+                "rawPredictionCol",
+                "weightCol",
+                "leafCol",
+            ],
+        )
+        params["seed"] = 1
+        params["eps"] = float
+        params["min_samples"] = int
```

Comment:
Do we need to set these params if they can be inferred from DBSCAN __init__?

Reply:
Let me check, since UMAP inspected from PCA I thought we need to inspect a spark algo